rfcs/0xx-tei_adapter/README.md

```diff
@@ -0,0 +1,93 @@
+# TEI Adapter
+
+## What is TEI
+## How we do it
+### Do we need an adapter?
+
+The TEI files are in a [GitHUb Repository](https://github.com/wellcomecollection/wellcome-collection-tei). 
+At the time of writing in April 2021 there's about 500(?) TEI files.
+We need to be able to propagate updates as they happen through to the API, and we also need to be able to
+reindex when we make changes in the pipeline. 
+
+GitHub is a relatively reliable and low latency source, so we could in theory just send the changes to a TEI transformer 
+and do a reindex by cloning the repository and/or crawling the TEI files. 
+There are three disadvantages to this approach:
+- We need a "special" reindexer for TEI.
+- We can't keep track of deleted files. If something is deleted, the change will go through the pipeline once, 
+  but there's no way of "remembering" it.
+- We need to assign versions to works resulting from TEI files in order to ensure subsequent updates are 
+  processed correctly in the pipeline. This versioning logic would have to be added explicitly to the 
+  TEI transformer whereas having an adapter and an intermediate store would allow us to use the already existing versioning
+  logic in the versioned store library.
+  
+For these reasons the proposal is to have a TEI adapter that listens to changes to TEI files and a TEI store.
+
+However, since GitHub is a relatively reliable and low latency source, the proposal is to only store the HTTP URLs
+to the TEI files in a TEI store, and not the full XML. This has the advantage that we don't have to manage an 
+ever growing S3 bucket (we don't have a way of cleaning up old versions of a record) but it has some disadvantages:
+- GitHub outages (which are rare but happen) might affect our reindexes.
+- We might be throttled during a reindex (?)
+- We go to the public internet for every TEI file for every reindex as opposed to getting each file exactly once 
+  per every change that affects it.
+  
+For these reasons, the proposal here is to store thw whole TEI file in a full VHS.
```

Comment:
I'm confused – L25 says the proposal is to only store the HTTP URLs, L33 says we're storing the whole TEI file.

I think you're trying to say "there are two approaches we could pick from, here is why approach #1 is bad, so we're going to use approach #2".

Reply:
sorry, my bad. Will fix thanks